src/main/java/net/minecraftforge/common/loot/IGlobalLootModifierSerializer.java

```diff
@@ -0,0 +1,17 @@
+package net.minecraftforge.common.loot;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.loot.conditions.ILootCondition;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+public interface IGlobalLootModifierSerializer<T extends IGlobalLootModifier> extends IForgeRegistryEntry<IGlobalLootModifierSerializer<?>> {
+    public abstract T read(ResourceLocation location, JsonObject object, ILootCondition[] ailootcondition);
+
+    @Override
+    public default Class<IGlobalLootModifierSerializer<?>> getRegistryType() {
```

Comment:
This needs a valid value.

Reply:
Good point. I looked at it a day or so ago (I saw that it was returning null and went "this probably should not do that") but I have been unable to determine what it *should* return. The existing implementations don't help (there being only two: one is `(Class<V>)token.getRawType();` and the other is `Serializer.<IRecipeSerializer<?>>castClass(IRecipeSerializer.class);`, attempting to mimic either one just generates compile-time errors).

Reply:
The one to mimic, due to generics, would be IRecipeSerializer. But you may have to do some finessing with generics to get it correct. Or, you could just make IGlobalLootModifierSerializer non-generic.

Reply:
Yeah, I was trying to play with the IRecipeSerializer one, but it kept fighting me. IRecipeSerializer contains within it a `Serializer` definition, which I don't have.

Reply:
Figured this one out. Involved moving the functionality down to the base class instead of in the interface `default`, but that's fine. There had been a part of the implementation in IRecipeSerializer that I overlooked.